src/Frame.js

```diff
@@ -0,0 +1,49 @@
+function Frame() {
+  this.score = 0;
```

Comment:
Avoid storing multiple versions of the truth.  Strikes, spares, completion and scores are all derived from the other properties - so these should all be functions that calculate the answers on the fly.